File: csharp/run-length-encoding/RunLengthEncoding.cs
using System.Text.RegularExpressions;

public static class RunLengthEncoding
{
    public static string Encode(string input)
    {
        return Regex.Replace(input, @"(.)\1+", match => $"{match.Length}{match.Value[0]}");
    }

    public static string Decode(string input)
    {
        return Regex.Replace(input, @"\d+.", match =>
            new string(
                match.Value[match.Value.Length - 1],
                int.Parse(match.Value.Substring(0, match.Value.Length - 1))
            )
         );
    }
}

File: javascript/run-length-encoding/run-length-encoding.js
export const encode = (string) => {
  return string.match(/(.)\1*/g)?.map(c => (c.length > 1 ? c.length.toString() : '') + c[0]).join('') ?? '';
};

export const decode = (string) => {
  return string.match(/\d*./g)?.map(c => c.length > 1 ? c[c.length - 1].repeat(c.substring(0, c.length - 1)) : c[0]).join('') ?? '';
};
